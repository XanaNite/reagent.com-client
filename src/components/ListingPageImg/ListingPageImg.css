.presentedBy{
    display: flex;
}

.presentedBy p{
    margin: 5px;
}

.listing-container{
    position: relative;
}

.listing-images img{
    width: 100%;
}

.imagesCount{
    position: absolute;
    bottom: 8px;
    left: 16px;
    color: white;
    background-color: black;
    opacity: 0.6;
    padding: 0px 5px 2px 5px;
    font-size: 12px;
}

.address h2{
    margin-bottom: 0;
}

.address p{
    margin: 0px;
    padding: 10px;
}

.listingAddress-details table{
    width: 100%;
}

.listingAddress-details td{
    width: 50%;
    text-align: left;
}

