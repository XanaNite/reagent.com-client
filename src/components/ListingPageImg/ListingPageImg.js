import React from 'react'

export default function ListingPageImg(){
    return(
        <section class="listing-about detailedBorder">
            <div class="presentedBy">
                <p>Presented by:</p>
                <p>[[AGENT]] with [[OFFICE]]</p>
            </div>
            <div class="listing-images">
                <div class="listing-container">
                    <img src={window.location.origin + "/images/placeholderHome.jpg"} alt="House" />
                    <div class="imagesCount">
                        <span>1/20</span>
                    </div>
                </div>
            </div>
            <div class="propertyAddress-details listingPage-details">
                <div class="address">
                    <h2 class="detailedBorder">[[COST]]</h2>
                    <div class="listing-info detailedBorder flexbox-around">
                        <p>bed</p>
                        <p>bath</p>
                        <p>sq ft</p>
                        <p>sq ft lot</p>
                    </div>
                    <p class="listingAddress detailedBorder">address</p>
                </div>
            </div>
            <div class="listingAddress-details">
                <table>
                    <tr>
                        <td>Property Type</td>
                        <td>[[INFO]]</td>
                    </tr>
                    <tr>
                        <td>Last Sold</td>
                        <td>[[INFO]]</td>
                    </tr>
                    <tr>
                        <td>Days on Market</td>
                        <td>[[INFO]]</td>
                    </tr>
                    <tr>
                        <td>Year Built</td>
                        <td>[[INFO]]</td>
                    </tr>
                    <tr>
                        <td>Price per sqft</td>
                        <td>[[INFO]]</td>
                    </tr>
                    <tr>
                        <td>Style</td>
                        <td>[[INFO]]</td>
                    </tr>
                </table>
            </div>
        </section>
    )
}
