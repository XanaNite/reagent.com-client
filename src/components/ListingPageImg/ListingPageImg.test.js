import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListingPageImg from './ListingPageImg';

describe(`ListingPageImg component`, () => {
    const props = {
        className: 'test-class-name',
        children: <p>test children</p>,
        'data-other': 'test-other-prop',
        agent: {
            first_name: "test",
            last_name: "test",
            title: "test",
            city: "test",
            state: "test"
        }
    };

    it('renders the ListingPageImg given props', () => {
        const wrapper = shallow(<ListingPageImg {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})