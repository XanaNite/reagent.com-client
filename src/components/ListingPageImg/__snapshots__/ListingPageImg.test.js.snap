// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingPageImg component renders the ListingPageImg given props 1`] = `
<section
  className="listing-about detailedBorder"
>
  <div
    className="presentedBy"
  >
    <p>
      Presented by:
    </p>
    <p>
      [[AGENT]] with [[OFFICE]]
    </p>
  </div>
  <div
    className="listing-images"
  >
    <div
      className="listing-container"
    >
      <img
        alt="House"
        src="http://localhost/images/placeholderHome.jpg"
      />
      <div
        className="imagesCount"
      >
        <span>
          1/20
        </span>
      </div>
    </div>
  </div>
  <div
    className="propertyAddress-details listingPage-details"
  >
    <div
      className="address"
    >
      <h2
        className="detailedBorder"
      >
        [[COST]]
      </h2>
      <div
        className="listing-info detailedBorder flexbox-around"
      >
        <p>
          bed
        </p>
        <p>
          bath
        </p>
        <p>
          sq ft
        </p>
        <p>
          sq ft lot
        </p>
      </div>
      <p
        className="listingAddress detailedBorder"
      >
        address
      </p>
    </div>
  </div>
  <div
    className="listingAddress-details"
  >
    <table>
      <tr>
        <td>
          Property Type
        </td>
        <td>
          [[INFO]]
        </td>
      </tr>
      <tr>
        <td>
          Last Sold
        </td>
        <td>
          [[INFO]]
        </td>
      </tr>
      <tr>
        <td>
          Days on Market
        </td>
        <td>
          [[INFO]]
        </td>
      </tr>
      <tr>
        <td>
          Year Built
        </td>
        <td>
          [[INFO]]
        </td>
      </tr>
      <tr>
        <td>
          Price per sqft
        </td>
        <td>
          [[INFO]]
        </td>
      </tr>
      <tr>
        <td>
          Style
        </td>
        <td>
          [[INFO]]
        </td>
      </tr>
    </table>
  </div>
</section>
`;
